### DISCUSSION TITLE
Generated Github release workflow does not build wheels for Python 3.8 - 3.10 for MacOS

### DISCUSSION THREAD

**Discussion by seddonym** [PyO3/maturin#2115] [Q&A] [answered]
### Bug Description

I recently switched to a Github release workflow generated by the newest version of Maturin (1.6.0), for my open source library Grimp. The release is missing wheels for 3.8 - 3.9 for macos arm64 and 3.8 - 3.10 for macos x86-64 ([issue on Grimp here](https://github.com/seddonym/grimp/issues/154)). 

I think this is possibly an issue with the setup-python action, which from what I can see is silently failing to install those versions. Possibly it is this (closed) issue on [setup-python](https://github.com/actions/setup-python/issues/808).

### Your maturin version (`maturin --version`)

1.6.0

### Your Python version (`python -V`)

n/a

### Your pip version (`pip -V`)

n/a

### What bindings you're using

None

### Does `cargo build` work?

- [ ] Yes, it works

### If on windows, have you checked that you aren't accidentally using unix path (those with the forward slash `/`)?

- [ ] Yes

### Steps to Reproduce

You can see the issue by looking at the CI run in question. https://github.com/seddonym/grimp/actions/runs/9497248945. Inside the 'Build wheels' actions under the Macos actions, you see it only finds some of the expected Python versions. Compare these with the linux and windows ones where it finds everything from 3.8 - 3.12.

I think you could reproduce the issue by setting up a new maturin project and then creating a Github release action using the docs. https://www.maturin.rs/distribution#github-actions. Let me know if you need me to go into more detail.

**Comment by messense** [comment]
> The release is missing wheels for 3.8 - 3.9 for macos arm64

There are no (official) support for Python < 3.11 on macOS arm64 so this is expected, for x86_64 it should work if you install them via `setup-python`.

**Reply by seddonym** [reply]
> There are no (official) support for Python < 3.11 on macOS arm64

Thanks for the reply. Do you mean no support from maturin, or from setup-python?

If it's the latter, I wonder if there is a way of maturin mitigating the issue somehow with an adjusted release.yml. Not supporting Python 3.10 on Mac feels to me kind of extreme, is that just me? 😄

**Reply by messense** [reply]
From CPython, only 3.11 and later versions have official python.org release

**Comment by seddonym** [comment, marked as answer]
I was able to get the release script working for earlier versions of Python on MacOS by [listing them explicitly](https://github.com/seddonym/grimp/pull/156/commits/cb065c24ddf5d6371919be9b6a1d3200030f6d8e), rather than in the `3.x` format specifed [here](https://github.com/PyO3/maturin/blob/844a2d608b1a3c9583743a1c45bc8586a0576ed3/src/ci.rs#L862).

```
     python-version: |
            3.8
            3.9
            3.10
            3.11
            3.12
```
